### DISCUSSION TITLE
How does luarocks solve different versons of dependencies?

### DISCUSSION THREAD

**Discussion by linrongbin16** [keplerproject/luarocks#1558] [Q&A]
For example, now I have two rocks `a` and `b`, they both reply on another rocks `c`, but in different versions, say `1.6.*`, and `2.*`.

This is extremely important when they use an api from `c`, and it's just the break changes from v1.x to v2.x.

How does luarocks solve this issue?

**Comment by alerque** [comment]
I don't believe it does automatically. I expect it to throw an error about being unable to resolve dependncies.

Your best bet is to fork `a`, update it to use the `c` v2 API, submit a pull request, then use your fork until it gets accepted.

Alternatively you can use the `--tree` argument to install `a` and `b` to different trees, then patch up either your usage of them or things in them so that they explicitly load the full path to those trees's version of `c` (so that `require('c')` doesn't just pull from the preload cache from whatever loaded a version first).
